### PR TITLE
feature:S3C-3532 Create user Implementation

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
* Implement Create User w/ all IAM APIs
  * Call AssumeRoleBackbeat to get temporary credentials
  * Call IAM Create User
  * Call IAM Get User Policy if exists
  * If Policy does not exist, Create User Policy
  * Create Access key for the user
 * Implemented Create User Test cases

### CODE REVIEW

**Review by anurag4DSB — DISMISSED** [review]

**Review by ilkescality — APPROVED** [review]

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java [hunk 1]
```diff
@@ -99,6 +110,54 @@ public static AssumeRoleRequest getAssumeRoleRequestForAccount(String accountID,
                 .withRoleSessionName(ROLE_SESSION_NAME_PREFIX + new Date().getTime());
     }
 
+    public static CreateUserRequest toCreateUserRequest(OsisUser osisUser) {
+        OsisUser.RoleEnum role = (osisUser.getRole() != null) ? osisUser.getRole() :OsisUser.RoleEnum.UNKNOWN ;
+        return new CreateUserRequest()
+                .withUserName(osisUser.getCdUserId())
+                .withPath(
+                        toUserPath(osisUser.getUsername(), role));
```
Comment (saitharunthodupunuru) [inline osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:118]: store email address too

Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java]: done

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java [hunk 2]
```diff
@@ -203,6 +213,55 @@ private static String toAdminPolicyDescription(String accountID) {
...
+        return new CreateUserRequest()
+                .withUserName(osisUser.getCdUserId())
+                .withPath(
+                        toUserPath(osisUser.getUsername(), role,
+                                osisUser.getEmail(), osisUser.getCdTenantId()));
+    }
+
+    private static String toUserPath(String username, OsisUser.RoleEnum role, String email, String cdTenantId) {
+        return "/" + username + "/" + role.getValue() + "/" + email + "/" + cdTenantId + "/" ;
+    }
```
Comment (anurag4DSB) [inline osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:227]: would you be able to add a review commit to the design doc?
Once we have the commit there, we can ask people who have reviewed already to review just that commit, and then we can merge this.

Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:227]: Here is the commit for the recent design changes:
c1cc56602b590be96bfda10223fd7da94132b909

**Comment by saitharunthodupunuru** [comment]
rebase done based on development branch

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java (hunk 2 above)
Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:227]: Merging this branch. Thanks

**Review by anurag4DSB — APPROVED** [review]

**Pull request merged**